sdk/core/azure-core/src/main/java/com/azure/core/http/HttpHeaders.java

```diff
@@ -66,6 +66,34 @@ public int getSize() {
         return headers.size();
     }
 
+    /**
+     * Adds a {@link HttpHeader header} with the given name and value if a header with that name doesn't already exist,
+     * otherwise adds the {@code value} to the existing header.
+     *
+     * @param name The name of the header.
+     * @param value The value of the header.
+     * @return The updated HttpHeaders object.
+     */
+    public HttpHeaders add(String name, String value) {
+        if (name == null) {
+            return this;
+        }
+        String caseInsensitiveName = formatKey(name);
+        if (value == null) {
```

Comment:
we can combine this with the first null check - `if(name == null || value == null) return this;`.  Saves formatting the name.